### PR TITLE
[IMP] l10n_ca: Add/Imp COA, taxes, reports

### PR DESCRIPTION
**Pull request by JulienAlardot**
Complete refactor of the module,
adding the tax report

task-3098316

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by JulienAlardot** [comment]
retarget of https://github.com/odoo/odoo/pull/128100

File: addons/l10n_ca/data/tax_report.xml [hunk 1]
```diff
@@ -0,0 +1,848 @@
...
+                elif CaBcB.balance <= 333.33 then 22
+                elif CaBcB.balance <= 3000 then CaBcB.balance * 6.6%
+                else: => 198
+                -->
+                <field name="hierarchy_level">3</field>
+                <field name="expression_ids">
+                    <record id="l10n_ca_tr_pst_bc_c_under_22" model="account.report.expression">
+                        <field name="label">under_22</field>
+                        <field name="engine">aggregation</field>
+                        <field name="formula">CaBcB.balance-22</field>
```
Comment (smetl) [inline addons/l10n_ca/data/tax_report.xml:238]:
```suggestion
                        <field name="formula">CaBcB.balance - 22</field>
```

File: addons/l10n_ca/data/tax_report.xml [hunk 2]
```diff
@@ -0,0 +1,848 @@
...
+                        <field name="subformula">if_below(CAD(0))</field>
+                    </record>
+                    <record id="l10n_ca_tr_pst_bc_c_under_333" model="account.report.expression">
+                        <field name="label">under_333</field>
+                        <field name="engine">aggregation</field>
+                        <field name="formula">(CaBcB.balance-333.33)*0.066</field>
```
Comment (smetl) [inline addons/l10n_ca/data/tax_report.xml:244]:
```suggestion
                        <field name="formula">(CaBcB.balance - 333.33) * 0.066</field>
```

File: addons/l10n_ca/data/tax_report.xml [hunk 3]
```diff
@@ -0,0 +1,848 @@
...
+                        <field name="subformula">if_above(CAD(0))</field>
+                    </record>
+                    <record id="l10n_ca_tr_pst_bc_c_over_3000" model="account.report.expression">
+                        <field name="label">over_3000</field>
+                        <field name="engine">aggregation</field>
+                        <field name="formula">-(CaBcB.balance-3000)*0.066</field>
```
Comment (smetl) [inline addons/l10n_ca/data/tax_report.xml:250]:
```suggestion
                        <field name="formula">- (CaBcB.balance - 3000) * 0.066</field>
```

File: addons/l10n_ca/data/tax_report.xml [hunk 4]
```diff
@@ -0,0 +1,848 @@
...
+                    <record id="l10n_ca_tr_pst_mb_2_under_200" model="account.report.expression">
+                        <field name="label">under_200</field>
+                        <field name="engine">aggregation</field>
+                        <field name="formula">(CaMb1.balance - 20) * 0.15</field>
+                        <field name="subformula">if_above(CAD(0))</field>
+                    </record>
+                    <record id="l10n_ca_tr_pst_mb_2_under_3000" model="account.report.expression">
+                        <field name="label">under_3000</field>
+                        <field name="engine">aggregation</field>
+                        <field name="formula">-(CaMb1.balance - 200 ) * 0.14</field>
```
Comment (smetl) [inline addons/l10n_ca/data/tax_report.xml:384]:
```suggestion
                        <field name="formula">-(CaMb1.balance - 200) * 0.14</field>
```

File: addons/l10n_ca/data/tax_report.xml [hunk 5]
```diff
@@ -0,0 +1,848 @@
...
+                    <record id="l10n_ca_tr_qst_205_aggr" model="account.report.expression">
+                        <field name="label">qst</field>
+                        <field name="engine">aggregation</field>
+                        <field name="formula">CaQc103203.qst + CaQc104204.qst</field>
+                    </record>
+                </field>
+            </record>
+            <record id="l10n_ca_tr_qst_106_206" model="account.report.line">
+                <field name="name">106 - 206 - ITCs/ITRs</field>
+                <field name="code">CaQc106206</field>
```
Comment (smetl) [inline addons/l10n_ca/data/tax_report.xml:509]: Such names are wtf :face_with_spiral_eyes:

File: addons/l10n_ca/data/res_company_data.xml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <odoo noupdate="1">
-<record model="res.company" id="base.main_company">
-    <field name="paperformat_id" ref="base.paperformat_us"/>
-</record>
+    <record model="res.company" id="base.main_company">
```
Comment (smetl) [inline addons/l10n_ca/data/res_company_data.xml:3]: Now we have a `demo_company_ca` company, this file is useless I think.
EDIT: Changing a demo data outside the `demo` folder seems to be a very bad idea...

File: addons/l10n_ca/demo/demo_company.xml [hunk 7]
```diff
@@ -24,12 +24,12 @@
 
     <function model="res.users" name="write">
         <value eval="[ref('base.user_root'), ref('base.user_admin'), ref('base.user_demo')]"/>
-        <value eval="{'company_ids': [(4, ref('l10n_ca.demo_company_ca'))]}"/>
+        <value eval="{'company_ids': [Command.link(ref('l10n_ca.demo_company_ca'))]}"/>
```
Comment (smetl) [inline addons/l10n_ca/demo/demo_company.xml:27]: need `paperformat_id` on demo_company_ca

File: addons/l10n_ca/models/res_partner.py [hunk 8]
```diff
@@ -1,8 +1,7 @@
-# coding: utf-8
-from odoo import api, fields, models, _
+from odoo import fields, models
 
 
 class ResPartner(models.Model):
     _inherit = 'res.partner'
 
-    l10n_ca_pst = fields.Char('PST')
+    l10n_ca_pst = fields.Char(string='PST number', help='Canadian Provincial Tax Identification Number')
```
Comment (smetl) [inline addons/l10n_ca/models/res_partner.py:7]: I think the "normal" way to handle that is to define a `vat_label` on the company

Comment (JulienAlardot) [reply addons/l10n_ca/models/res_partner.py:7]: Yes but PST isn't VAT, and it's a different number than the VAT one

File: addons/l10n_ca/data/tax_report.xml (hunk 5 above)
Comment (JulienAlardot) [reply addons/l10n_ca/data/tax_report.xml:509]: They make sense, but I admit it's an acquired taste :p

**Review by smetl — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**